### ISSUE TITLE
Feature request: Client capabilities exchange

### ISSUE THREAD

**Issue by fling-** [TokTok/c-toxcore#888]
Clients should exchange with capabilities packets when negotiating a connection.
The packet should contain GUIDs of the features supported by the client.
Example list:
```
chat.tox:some-feature
io.utox:another-one
```
Sending toxcore version is not enough because the clients might not support all the features implemented in toxcore.
Just an integer value of the client feature level height should not be used as the development is decentralized and there is a lot of clients implementing support of different things in parallel. Per capability/feature GUID should be suitable for the task.

**Comment by Diadlo** [comment]
Seems, duplicate of #513, so it will be closed. Please comment if it isn't

**Comment by fling-** [comment]
@Diadlo I missed this one, sorry.
